bootstore/src/db/mod.rs

```diff
@@ -76,41 +67,95 @@ impl Db {
         // Create tables
         c.batch_execute(&schema)?;
 
-        Ok(Db { log, conn: c })
+        Ok((Db { log, path: path.to_string() }, c))
+    }
+
+    pub fn new_connection(&self) -> Result<SqliteConnection, Error> {
```

Comment:
This makes me a little nervous, because it opens the door for concurrency issues between multiple connections to the same database. If I understood the concurrency plan in the PR description, there will be just one database connection running in its own thread - why did this get split out instead of `Db` continuing to hold the one and only connection?

Reply:
> This makes me a little nervous, because it opens the door for concurrency issues between multiple connections to the same database. 

This isn't really an issue, as everything will be wrapped in a transaction and is the whole point of using multiple connections. We wouldn't use any application read caching in this case, as reads are concurrent on a  WAL enabled sqlite DB. However, as you surmised, there is still care that has to be taken and its more likely for transactions to be aborted in this case, which could leave open unhandled errors in production. This along with the fact that write transactions execute one at a time and we persist them all for safety, leads me to the design I proposed where we used a single connection.

> If I understood the concurrency plan in the PR description, there will be just one database connection running in its own thread - why did this get split out instead of `Db` continuing to hold the one and only connection?

I had made that change before I decided upon that strategy 😅  It was either that or  use a mutex or cell since I was getting multiple mutable borrow check errors.  I'll very likely go back to having `Db` hold the sole connection and remove the extra parameter from every call.

Reply:
Yeah, that's my chief concern as well, besides added complexity. I'll go towards the single connection plan and put it back in `Db`.

Reply:
Done!